src/Dev/Tasks/TagsToShortcodeHelper.php

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /**
+     * TagsToShortcodeHelper constructor.
+     * @param string $baseClass The base class that will be used to look up HTMLText fields
+     * @param bool $includeBaseClass Whether to include the base class' HTMLText fields or not
+     */
+    public function __construct($baseClass = null, $includeBaseClass = false)
+    {
+        $flysystemAssetStore = singleton(AssetStore::class);
+        if (!($flysystemAssetStore instanceof FlysystemAssetStore)) {
+            throw new InvalidArgumentException("FlysystemAssetStore missing");
+        }
+        $this->flysystemAssetStore = $flysystemAssetStore;
+
+        $this->baseClass = $baseClass ?: DataObject::class;
+        $this->includeBaseClass = $includeBaseClass;
+
+        $this->validTagsPattern = implode('|', array_keys(static::VALID_TAGS));
+        $this->validAttributesPattern = implode('|', static::VALID_ATTRIBUTES);
+    }
+
+    /**
+     * @throws \ReflectionException
+     */
+    public function run()
+    {
+        Environment::increaseTimeLimitTo();
+
+        $classes = DataObjectSchema::getFieldMap($this->baseClass, $this->includeBaseClass, ['HTMLText', 'HTMLVarchar']);
+        foreach ($classes as $class => $tables) {
+            foreach ($tables as $table => $fields) {
+                foreach ($fields as $field) {
+                    try {
+                        $sqlSelect = SQLSelect::create(
+                            ['ID', $field],
+                            $table
+                        );
+                        $whereAnys = [];
+                        foreach (array_keys(static::VALID_TAGS) as $tag) {
+                            $whereAnys[]= "\"$table\".\"$field\" LIKE '%<$tag%'";
+                            $whereAnys[]= "\"$table\".\"$field\" LIKE '%[$tag%'";
+                        }
+                        $sqlSelect->addWhereAny($whereAnys);
+                        $records = $sqlSelect->execute();
+                        $this->rewriteFieldForRecords($records, $table, $field);
+                    } catch (DatabaseException $exception) {
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Takes a set of query results and updates image urls within a page's content.
+     * @param Query $records
+     * @param string $updateTable
+     * @param string $field
+     */
+    private function rewriteFieldForRecords(Query $records, $updateTable, $field)
+    {
+        foreach ($records as $row) {
+            $content = $row[$field];
+            $newContent = $this->getNewContent($content);
+            if ($content == $newContent) {
+                continue;
+            }
+
+            $updateSQL = SQLUpdate::create($updateTable)->addWhere(['"ID"' => $row['ID']]);
+            $updateSQL->addAssignments(["\"$field\"" => $newContent]);
+            $updateSQL->execute();
+            if ($this->logger) {
+                $this->logger->info("Updated page with ID {$row['ID']}");
```

Comment:
Pages aren't the only things with HTMLText fields - this needs to mention the record type